roles/rhscl-postgresql-apb-openshift/tasks/main.yml

```diff
@@ -1,3 +1,25 @@
+- block:
+  - name: Find pod we need to update
+    shell: oc get pods -n {{ namespace }} -l app=rhscl-postgresql-apb -o custom-columns=name:.metadata.name --no-headers
+    register: oldpod
+
+  - name: Find dc we will clean up
+    shell: oc get dc -n {{ namespace }} -l app=rhscl-postgresql-apb -o custom-columns=name:.metadata.name --no-headers
+    register: olddc
+
+  - name: Create source pgpass
+    shell: oc exec -n {{ namespace }} {{ oldpod.stdout }} -- /bin/bash -c "echo '*:*:*:{{ postgresql_user }}:{{ postgresql_password }}' > ~/.pgpass"
+
+  - name: Set permissions on source pgpass
+    shell:  oc exec -n {{ namespace }} {{ oldpod.stdout }} -- /bin/bash -c "chmod 600 ~/.pgpass"
+
+  - name: Backup source database
+    shell:  oc exec -n {{ namespace }} {{ oldpod.stdout }} -- /bin/bash -c "PGPASSFILE=~/.pgpass pg_dump -h localhost -U '{{ postgresql_user }}' '{{ postgresql_database }}' > /tmp/db.dump"
+
+  - name: Copy over db backup
+    shell: oc cp -n {{ namespace }} {{ oldpod.stdout }}:/tmp/db.dump /tmp/db.dump
+  when: update is defined
```

Comment:
is `when: update` not sufficient?

Reply:
I ask that because you set `update: true` in the update playbook.

Reply:
if you do when: update and it is not defined (provision, deprovision, etc.) I _think_ it will bomb.